addons/web_editor/static/src/js/editor/snippets.editor.js

```diff
@@ -1112,6 +1113,13 @@ var SnippetEditor = Widget.extend({
             this.$el.css('pointer-events', '');
         }, 250);
     },
+    /**
+     * @private
+     * @param {OdooEvent} ev
+     */
+    _onReplaceTarget: function (ev) {
+        this.destroy();
+    },
```

Comment:
I don't like the Idea of destroying the SnippetEditor if the image is just replaced, wouldn't we want something more graceful? Maybe the SnippetMenu could be made aware that there was a change in the DOM and that it needs to re-evaluate its SnippetsEditor, what do you think?

Right now if I replace the image I end up with an empty option tab, where as I think I should just have the image options be gone.

Reply:
Now running the `destroy()` through the mutex of `SnippetsMenu`.